examples/scenarioSmallBodyLandmarks.py

```diff
@@ -0,0 +1,571 @@
+                    triangles=order_face-1, color=color_asteroid, zorder=0, alpha=0.1)
+    ax.plot(posLmk[idxOn, 0] / 1000, posLmk[idxOn, 1] / 1000, posLmk[idxOn, 2] / 1000, 'b', linestyle='',
+            marker='.', markersize=5)
+    ax.plot(posLmk[idxOff, 0] / 1000, posLmk[idxOff, 1] / 1000, posLmk[idxOff, 2] / 1000, 'r', linestyle='',
+            marker='.', markersize=5, alpha=0.25)
+    ax.quiver(r[0] / 1000, r[1] / 1000, r[2] / 1000, dcm_CP[0,0], dcm_CP[1,0], dcm_CP[2,0], length=10, normalize=True,
+              color='black', alpha=0.25)
+    ax.quiver(r[0] / 1000, r[1] / 1000, r[2] / 1000, dcm_CP[0,1], dcm_CP[1,1], dcm_CP[2,1], length=10, normalize=True,
+              color='black', alpha=0.25)
+    ax.quiver(r[0] / 1000, r[1] / 1000, r[2] / 1000, dcm_CP[0,2], dcm_CP[1,2], dcm_CP[2,2], length=10, normalize=True,
+              color='blue')
+    ax.set_xlabel('${}^{P}r_{x}$ [km]')
+    ax.set_ylabel('${}^{P}r_{y}$ [km]')
+    ax.set_zlabel('${}^{P}r_{z}$ [km]')
+    t_str = str(int(t/60))
+    ax.set_title('Asteroid rotating frame, t=' + t_str + ' min')
+
+
+def plot_pixel(pixelLmk, statusLmk):
+    """Plot landmarks labeled pixels in the camera image."""
+    idxOn = statusLmk == 1
+    pixelOn = pixelLmk[idxOn, :]
+
+    fig = plt.figure()
+    ax = fig.gca()
+    plt.plot(pixelOn[:,0], pixelOn[:,1], linestyle='', marker='s', color='b', markersize=5)
+    for i in range(len(pixelOn)):
+        ax.annotate(str(i), (pixelOn[i,0], pixelOn[i,1]))
+    ax.set_xlim([-1024, 1024])
+    ax.set_ylim([-768, 768])
+    plt.xlabel('$p_x$ [-]')
+    plt.ylabel('$p_y$ [-]')
+
+
+def plot_nLmk(t, nvisibleLmk):
+    """Plot visible landmarks evolution."""
+    fig = plt.figure()
+    ax = fig.gca()
+    plt.plot(t/3600, nvisibleLmk, linestyle='--', marker='.', markersize=8)
+    ax.set_xlim([t[0]/3600, t[-1]/3600])
+    plt.xlabel('Time [h]')
+    plt.ylabel('Visible landmarks [-]')
+
+
+def plot_orientation(t, dcm_HP, dcm_CP):
+    """Plot the camera frame orientation with respect to Hill frame."""
+    data = np.zeros((len(t), 3))
+    for i in range(len(t)):
+        data[i,0:3] = [np.dot(dcm_HP[i, 0:3, 0], dcm_CP[i, 0:3, 2]),
+                       np.dot(dcm_HP[i, 0:3, 1], dcm_CP[i, 0:3, 0]),
+                       np.dot(dcm_HP[i, 0:3, 2], dcm_CP[i, 0:3, 1])]
+    fig = plt.figure()
+    ax = fig.gca()
+    labelStrings = (r'$\hat\imath_r\cdot \hat c_3$'
+                    , r'${\hat\imath}_{\theta}\cdot \hat c_1$'
+                    , r'$\hat\imath_h\cdot \hat c_2$')
+    for idx in range(3):
+        plt.plot(t/3600, data[:, idx],
+                 color=unitTestSupport.getLineColor(idx, 3),
+                 label=labelStrings[idx])
+    ax.set_xlim([t[0]/3600, t[-1]/3600])
+    plt.legend(loc='lower right')
+    plt.xlabel('Time [h]')
+    plt.ylabel('Orientation Illustration')
+
+
+def run(show_plots, useAltBodyFrame=True, useBatch=False, useGravPoly=False):
```

Comment:
Your run() argument takes several optional arguments.  But, you include this scenario in `test_scenarioTest.py` which only calls run() without arguments.  Thus, many versions of this code don't get tested.  Please write a dedicated `test_scenarioSmallBodyLandmark.py` file that parametrizes these options to ensure you run all combinations of this scenario.

Reply:
Thanks for clarifying this. I've tailored the arguments for that purpose and created a test_scenarioSmallBodyLandmark.py. It checks plots and that the Simulation and method provides same results